addons/website/static/tests/builder/website_helpers.js

```diff
@@ -0,0 +1,520 @@
+import { after, before, describe } from "@odoo/hoot";
+import {
+    advanceTime,
+    animationFrame,
+    click,
+    queryOne,
+    tick,
+    waitFor,
+    waitForNone,
+} from "@odoo/hoot-dom";
+import {
+    contains,
+    defineModels,
+    getService,
+    mockService,
+    models,
+    mountWithCleanup,
+    onRpc,
+    patchWithCleanup,
+} from "@web/../tests/web_test_helpers";
+import { loadBundle } from "@web/core/assets";
+import { isBrowserFirefox } from "@web/core/browser/feature_detection";
+import { registry } from "@web/core/registry";
+import { uniqueId } from "@web/core/utils/functions";
+import { WebClient } from "@web/webclient/webclient";
+import { patchWithCleanupImg } from "@html_builder/../tests/helpers";
+import { getWebsiteSnippets } from "./snippets_getter.hoot";
+import { mockImageRequests } from "./image_test_helpers";
+
+class Website extends models.Model {
+    _name = "website";
+    get_current_website() {
+        return [1];
+    }
+}
+
+class IrUiView extends models.Model {
+    _name = "ir.ui.view";
+    render_public_asset() {
+        return getWebsiteSnippets();
+    }
+}
+
+export const exampleWebsiteContent = '<h1 class="title">Hello</h1>';
+
+export const invisibleEl =
+    '<div class="s_invisible_el o_snippet_invisible" data-name="Invisible Element" data-invisible="1"></div>';
+
+export const wrapExample = `<div id="wrap" data-oe-model="ir.ui.view" data-oe-id="539" data-oe-field="arch">${exampleWebsiteContent}</div>`;
+
+export function defineWebsiteModels() {
+    describe.current.tags("desktop");
+    defineMailModels();
+    defineModels([Website, IrUiView]);
+    before(() => {
+        onRpc("/website/theme_customize_data_get", () => []);
+    });
+}
+
+/**
+ * This helper will be moved to website. Prefer using setupHTMLBuilder
+ * for builder-specific tests
+ */
+export async function setupWebsiteBuilder(
+    websiteContent,
+    {
+        snippets,
+        openEditor = true,
+        loadIframeBundles = false,
+        loadAssetsFrontendJS = false,
+        hasToCreateWebsite = true,
+        versionControl = false,
+        styleContent,
+        headerContent = "",
+        beforeWrapwrapContent = "",
+    } = {}
+) {
+    // TODO: fix when the iframe is reloaded and become empty (e.g. discard button)
+    if (hasToCreateWebsite) {
+        const pyEnv = await startServer();
+        pyEnv["website"].create({});
+    }
+    mockImageRequests();
+    registry.category("services").remove("website_edit");
+    let editor;
+    let editableContent;
+    await mountWithCleanup(WebClient);
+    let originalIframeLoaded;
+    let resolveIframeLoaded = () => {};
+    const iframeLoaded = new Promise((resolve) => {
+        resolveIframeLoaded = (el) => {
+            const iframe = el;
+            if (styleContent) {
+                const style = iframe.contentDocument.createElement("style");
+                style.innerHTML = styleContent;
```

Comment:
I believe style can/should always be set as `textContent` instead, as we never want to insert actual HTML into that tag.